src/order/directed.lean

```diff
@@ -189,6 +214,27 @@ lemma is_bot_iff_is_min [is_directed α (≥)] : is_bot a ↔ is_min a :=
 
 lemma is_top_iff_is_max [is_directed α (≤)] : is_top a ↔ is_max a := ⟨is_top.is_max, is_max.is_top⟩
 
+/-- The property that a function preserves `is_lub` on directed sets. -/
+def preserves_lub_on_directed [preorder β] (f : α → β) : Prop :=
+∀ (d : set α) (a : α), d.nonempty → directed_on (≤) d → is_lub d a → is_lub (f '' d) (f a)
```

Comment:
Can you define the dual notion?

Reply:
@YaelDillies The dual of an (upward) directed set is a downward directed set (sometimes referred to as a filtered set). Is this definition in Mathlib at the moment?

Reply:
So the dual notion would be:
```
def dual_scott_continuous [preorder β] (f : α → β) : Prop :=
∀ ⦃d : set α⦄, d.nonempty → directed_on (≥) d → ∀ ⦃a⦄, is_glb d a → is_glb (f '' d) (f a)
```
However, I'm not sure if it's any use?

In some circumstances the dual of the Scott topology is the lower topology, but this is not always true https://www.sciencedirect.com/science/article/pii/S0166864115005374

I should say that, beyond reading parts of Gierz et al whilst researching my D.Phil thesis twenty(!) years ago, I am not an expert in this area.

My inclination would be not to include the dual concept in Mathlib unless someone has a need for it.

Reply:
Can you leave a note to this effect, then?

Reply:
Done.